src/common/otel/metrics/taskMetrics.ts

```diff
@@ -0,0 +1,166 @@
+// src/common/telemetry/metrics/taskMetrics.ts
+import { singleton, inject } from 'tsyringe';
+import { Registry, Counter, Histogram, Gauge } from 'prom-client';
+import { OperationStatus } from '@map-colonies/mc-priority-queue';
+import { Logger } from '@map-colonies/js-logger';
+import { IConfig } from 'config';
+import { SERVICES } from '../../constants';
+
+export interface TaskMetricLabels {
+  jobType: string;
+  taskType: string;
+  status?: OperationStatus;
+  errorType?: string;
+}
+
+@singleton()
+export class TaskMetrics {
+  // Task processing metrics
+  private tasksProcessedCounter?: Counter;
+  private tasksProcessingDuration?: Histogram;
+  private tasksSuccessCounter?: Counter;
+  private tasksFailureCounter?: Counter;
+
+  // Gauge metrics
+  private activeTasksGauge?: Gauge;
+
+  // Configuration
+  private readonly taskBuckets: number[];
+  private readonly metricsEnabled: boolean;
+
+  public constructor(
+    @inject(SERVICES.CONFIG) private readonly config: IConfig,
+    @inject(SERVICES.LOGGER) private readonly logger: Logger,
+    @inject(SERVICES.METRICS_REGISTRY) private readonly registry?: Registry
+  ) {
+    this.metricsEnabled = this.config.get<boolean>('telemetry.metrics.enabled');
+    this.taskBuckets = this.config.get<number[]>('telemetry.metrics.buckets');
+
+    if (this.registry && this.metricsEnabled) {
+      this.initializeMetrics();
+      this.logger.info('Task metrics initialized');
+    }
+  }
```

Comment:
this whole class is based on the availability of registry
it is checked for existance in each function
i would suggest moving this logic one step up
initialize this class if metrics are enabled